### PR TITLE
Fix zlib CVE-2023-45853 in cloud-hypervisor

### PR DESCRIPTION
**Pull request by xordux**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix CVE-2023-45853

###### Change Log  

- Fix zlib CVE-2023-45853 in cloud-hypervisor

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-45853

###### Test Methodology

- Pipeline build id: [441558](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=441558&view=results)

### CODE REVIEW

**Comment by Neustradamus** [comment]
Linked to:
- https://github.com/madler/zlib/issues/868

**Comment by xordux** [comment]
**Note:** Everything is passing in buddy build except for installation on arm64 because cloud-hypervisor is not available in arm64.

**Review by mfrw — APPROVED** [review]
LGTM

**Review by nicogbg — APPROVED** [review]

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 1]
```diff
@@ -71,6 +72,9 @@ Cloud Hypervisor is an open source Virtual Machine Monitor (VMM) that runs on to
 %setup -q -n %{name}-%{version}
 %if 0%{?using_vendored_crates}
 tar xf %{SOURCE1}
+pushd vendor/libz-sys/src/zlib
```
Comment (jslobodzian) [inline SPECS/cloud-hypervisor/cloud-hypervisor.spec:75]: I don't think this is actually a dependency.  The build logs do not show zlib being compiled or libz being linked.

I think we might want to remove this from the vendor tarball though.

**Review by romoh — APPROVED** [review]

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec (hunk 1 above)
Comment (jslobodzian) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec:75]: This was included in the build's test infrastructure. Per Rohit:

_It turns out that cloud-hypervisor's dependency on libz is like this: test_infra -> ssh2 -> libssh2-sys -> libz-sys 
 
test_infra is part of cloud-hypervisor's build dependency [[link](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/42ca292dce4292f0f69bd7cecd7db09eee3b64bb/Cargo.toml#L91C18-L91C18)] which needs ssh2 [[link](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/42ca292dce4292f0f69bd7cecd7db09eee3b64bb/test_infra/Cargo.toml#L14C1-L14C1)]_

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**